lib/lotus/commands/generate/abstract.rb

```diff
@@ -0,0 +1,75 @@
+require 'lotus/environment'
+require 'thor'
+
+module Lotus
+  module Commands
+    class Generate < Thor
+      class Abstract
```

Comment:
Sure! I'll explain the _specifically one thing_ stuff and then put more thoughts regarding hierarchy :) Please, question if anything is dubious or wrong or whatever ;)

### Doing only one thing

I'm specifically talking about [`DatabaseConfig`](https://github.com/lotus/lotus/pull/325/files#diff-38afc003617f0593d1a62c002f6a8f42R3) class, I extracted common logic from `App` and `Container` classes and promoted to a new _player_ that does specifically one thing: Outputs database configs depending on an engine.

I had the option to simply promote this logic to the `Abstract` superclass, although, it'll end up as a bag of methods doing a bunch of things (handling database config, test framework's config, etc).

It's important to noticed that it's isn't 100% _done_ yet, `App` and `Container` stills references `DatabaseConfig` explicitly and knows how to build that dependency. I didn't want to go too far in this refactor, though.

So, my suggestion to share this common behaviour it extract all this test framework logic in a player that does that, perhaps `Lotus::Generators::TestFramework` :)


### Thoughts about hierarchy

Looking up the whole picture I found out another issue that may suggest that inheritance is possible not a good choice:
<img width="690" alt="screen shot 2015-08-23 at 4 19 05 pm" src="https://cloud.githubusercontent.com/assets/379894/9429814/c0e77b20-49b3-11e5-9501-ee983aaedb79.png">

A bunch of guys inherit from `Abstract` which characterises a wide hierarchy  shape:

![hierarchies_shapes](https://cloud.githubusercontent.com/assets/379894/9429829/21dcb792-49b4-11e5-856d-a0b5a25332ff.jpg)

Which means that `Abstract` class will be possibly harder to maintain having big churn and may end up with undesired and unpredictable collateral effects.